modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/APMMetric.java

```diff
@@ -0,0 +1,133 @@
+    private final Instruments instruments = new Instruments();
+    private volatile boolean enabled;
+    private AtomicReference<APMServices> services = new AtomicReference<>();
+
+    record APMServices(Meter meter, OpenTelemetry openTelemetry) {}
+
+    // TODO remove duplication between APMTracer and APMMetric. enabled, create apm services etc
+    public APMMetric(Settings settings) {
+        this.enabled = APM_ENABLED_SETTING.get(settings);
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        if (enabled) {
+            createApmServices();
+        } else {
+            destroyApmServices();
+        }
+    }
+
+    @Override
+    protected void doStart() {
+        if (enabled) {
+            createApmServices();
+        }
+    }
+
+    @Override
+    protected void doStop() {
+        destroyApmServices();
+    }
+
+    @Override
+    protected void doClose() {}
+
+    @Override
+    public <T> DoubleCounter registerDoubleCounter(MetricName name, String description, T unit) {
+        return instruments.registerDoubleCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleUpDownCounter registerDoubleUpDownCounter(MetricName name, String description, T unit) {
+        return instruments.registerDoubleUpDownCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleGauge registerDoubleGauge(MetricName name, String description, T unit) {
+        return instruments.registerDoubleGauge(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleHistogram registerDoubleHistogram(MetricName name, String description, T unit) {
+        return instruments.registerDoubleHistogram(name, description, unit);
+    }
+
+    @Override
+    public <T> LongCounter registerLongCounter(MetricName name, String description, T unit) {
+        return instruments.registerLongCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> LongUpDownCounter registerLongUpDownCounter(MetricName name, String description, T unit) {
+        return instruments.registerLongUpDownCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> LongGauge registerLongGauge(MetricName name, String description, T unit) {
+        return instruments.registerLongGauge(name, description, unit);
+    }
+
+    @Override
+    public <T> LongHistogram registerLongHistogram(MetricName name, String description, T unit) {
+        return instruments.registerLongHistogram(name, description, unit);
+    }
+
+    void createApmServices() {
+        assert this.enabled;
+        assert this.services.get() == null;
+
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            var openTelemetry = GlobalOpenTelemetry.get();
+            var meter = openTelemetry.getMeter("elasticsearch");
+
+            this.services.set(new APMServices(meter, openTelemetry));
```

Comment:
this is based on how tracer does it, but I wonder if we really need both?
we use only meter after all?